### PR TITLE
FLINK-3179 - Adding just a log message

### PR DESCRIPTION
**Pull request by ramkrish86**
As suggested in the JIRA, just adding a log message indicating to add an combiner function before the partition operator.

### CODE REVIEW

File: flink-optimizer/src/main/java/org/apache/flink/optimizer/operators/GroupReduceWithCombineProperties.java [hunk 1]
```diff
@@ -90,6 +94,9 @@ public DriverStrategy getStrategy() {
 	@Override
 	public SingleInputPlanNode instantiate(Channel in, SingleInputNode node) {
 		if (in.getShipStrategy() == ShipStrategyType.FORWARD) {
+			if(in.getSource().getOptimizerNode() instanceof PartitionNode) {
+				LOG.warn("Consider adding an explicit CombinerFunction with groupCombine in front of the partition operator");
```
Comment (fhueske) [inline flink-optimizer/src/main/java/org/apache/flink/optimizer/operators/GroupReduceWithCombineProperties.java]: Can you extend the error message as follows?

> Cannot automatically inject combiner for GroupReduceFunction. Please add an explicit combiner with combineGroup() in front of the partition operator.

File: flink-optimizer/src/main/java/org/apache/flink/optimizer/operators/ReduceProperties.java [hunk 2]
```diff
@@ -61,6 +65,9 @@ public SingleInputPlanNode instantiate(Channel in, SingleInputNode node) {
 		if (in.getShipStrategy() == ShipStrategyType.FORWARD ||
 				(node.getBroadcastConnections() != null && !node.getBroadcastConnections().isEmpty()))
 		{
+			if(in.getSource().getOptimizerNode() instanceof PartitionNode) {
+				LOG.warn("Consider adding an explicit CombinerFunction with groupCombine in front of the partition operator");
```
Comment (fhueske) [inline flink-optimizer/src/main/java/org/apache/flink/optimizer/operators/ReduceProperties.java]: Can you extend the error message as follows?

> Cannot automatically inject combiner for ReduceFunction. Please add an explicit combiner with combineGroup() in front of the partition operator.

**Comment by fhueske** [comment]
Thanks for the PR! Just had a suggestion for the wording of the log message. Looks good otherwise.

**Comment by ramkrish86** [comment]
Done the update @fhueske . Pls review. Thanks.

**Comment by fhueske** [comment]
Thanks for the update! Good to merge :-)

**Comment by fhueske** [comment]
Merging

**Pull request closed without merging**